### PR TITLE
Add release instructions

### PR DESCRIPTION
**Pull request by zxiiro**
Add instructions on how to produce a Sceptre release.

Signed-off-by: Thanh Ha <zxiiro@gmail.com>

[Your PR description here]

## PR Checklist

- [ ] Wrote a good commit message & description [see guide below].
- [ ] Commit message starts with `[Resolve #issue-number]`.
- [ ] Added/Updated unit tests.
- [ ] Added/Updated integration tests (if applicable).
- [ ] All unit tests (`make test`) are passing.
- [ ] Used the same coding conventions as the rest of the project.
- [ ] The new code passes pre-commit validations (`pre-commit run --all-files`).
- [ ] The PR relates to _only_ one subject with a clear title.
      and description in grammatically correct, complete sentences.

## Approver/Reviewer Checklist

- [ ] Before merge squash related commits.

## Other Information

[Guide to writing a good commit](http://chris.beams.io/posts/git-commit/)

### CODE REVIEW

**Comment by tarkatronic** [comment]
Should this maybe be added as part of the `CONTRIBUTING.md` doc? For example, I wrote this up for `tartufo`: https://tartufo.readthedocs.io/en/stable/CONTRIBUTING.html#contributing-as-a-maintainer

**Review by zaro0508 — APPROVED** [review]
LGTM, i'm fine with either a link from CONTRIBUTING.md or directly in CONTRIBUTING.md

**Comment by zxiiro** [comment]
> Should this maybe be added as part of the `CONTRIBUTING.md` doc? For example, I wrote this up for `tartufo`: https://tartufo.readthedocs.io/en/stable/CONTRIBUTING.html#contributing-as-a-maintainer

My thought for the CONTRIBUTING file is its instructions for non-maintainers who want to propose something to the project. I feel that polluting that file with things they don't care about might make it seem overwhelming to a new maintainer to read.

I think having a separate file for maintainers makes sense and maybe RELEASING might not be the right word but we should keep it separate in order to keep CONTRIBUTING file simple.

File: RELEASING.md
Comment (jfalkenstein) [inline RELEASING.md:3]: I think you should give a full filepath for `__init__.py`, since it's not immediately clear which one that refers to.

File: RELEASING.md
Comment (jfalkenstein) [inline RELEASING.md:4]: Probably more specific instructions for what to include could be very useful.

File: RELEASING.md
Comment (jfalkenstein) [inline RELEASING.md:8]: This seems really vague. Who is being emailed? What would the content of this email be?

**Comment by alexharv074** [comment]
fwiw, I wrote this bash script that could be adapted for releasing our fork 

```bash
#!/usr/bin/env bash

set -o errexit

usage() {
  echo "Usage: $0 VERSION [PART]"
  echo "e.g. $0 2.10.0"
  echo "e.g. $0 3.0.0 major"
  exit 1
}

get_opts() {
  [ -z "$1" ] && usage
  version="$1"
  part="minor"
  [ -n "$2" ] && part="$2" 
}

validate_opts() {
  if ! grep -Eq '^\d+\.\d+\.\d+$' <<< "$version" ; then
    echo "Version '$version' looks wrong"
    usage
  fi

  if ! grep -wq "$part" <<< "major minor" ; then
    echo "Part '$part' looks wrong, should be major or minor"
    usage
  fi
}

release() {
  local branch_name="release-$(tr -d "." <<< "$version")"

  git co -b "$branch_name"
  bumpversion --new-version "$version" "$part"
  git push origin "$branch_name"

  echo "\
Now raise a PR:
https://github.com/fsa-streamotion/sceptre/compare/\
develop...fsa-streamotion:$branch_name?expand=1"
}

main() {
  get_opts "$@"
  validate_opts
  release
}

if [ "$0" == "${BASH_SOURCE[0]}" ] ; then
  main "$@"
fi
```

Before that, I had these notes in our `RELEASE.md`:

> ## Bump version
>
> ... Note that this step can be automated using the `bumpversion` tool, which will be already installed in the virtualenv. For example:
> 
> ```text
> ▶ bumpversion --new-version 2.11.0 minor
> ```
>
> This will update the version strings in the source code and commit.

File: RELEASING.md
Comment (zxiiro) [reply RELEASING.md:3]: Done.

File: RELEASING.md
Comment (zxiiro) [reply RELEASING.md:4]: Done.

File: RELEASING.md
Comment (zxiiro) [reply RELEASING.md:8]: Done.

**Comment by zxiiro** [comment]
> fwiw, I wrote this bash script called `bump_version.sh` that could be adapted for bumping the version.

Thanks Alex, I think we can use this as a base to turn this into a GitHub Action that performs the release when a Git tag is pushed so the entire process is automated. Getting a generic RELEASE.md instruction step list is a good first step to documenting the essentials before adding the automation. I always like to have both to capture the essance of what is needed to be performed so that folks in the future can migrate CI/CD platforms if they need to for some reason.

**Review by zaro0508 — APPROVED** [review]

**Review by jfalkenstein — APPROVED** [review]

File: RELEASE.md
Comment (jfalkenstein) [inline RELEASE.md:8]: @zxiiro Is this done AFTER the squash-merge to master or before?

Comment (zxiiro) [reply RELEASE.md:8]: It should be done after merge so I guess we should have a step for merging and then doing a final `git pull`. Let me add that.

**Review by jfalkenstein — APPROVED** [review]

**Pull request merged**
